### PR TITLE
Add support for `flux::ignore` pragma to skip analyzing a module or whole crate

### PR DESCRIPTION
**Pull request by ranjitjhala**
See example tests

Pos

* [00](flux-tests/tests/neg/surface/ignore00.rs)

Neg

* [00](flux-tests/tests/neg/surface/ignore00.rs)
* [01](flux-tests/tests/neg/surface/ignore01.rs)

### CODE REVIEW

File: flux-driver/src/collector.rs [hunk 1]
```diff
@@ -27,12 +27,19 @@ pub(crate) struct SpecCollector<'tcx, 'a> {
     error_guaranteed: Option<ErrorGuaranteed>,
 }
 
+/// An IgnoreKey is either `Some(module_def_id)` or `None` indicating the entire crate
+pub type IgnoreKey = Option<LocalDefId>;
```
Comment (nilehmann) [inline flux-driver/src/collector.rs:31]: Maybe better to have an explicit enum

```suggestion
#[derive(PartialEq, Eq, Hash)]
pub enum IgnoreKey {
    Crate,
    Module(LocalDedIf)
}
```

File: flux-driver/src/collector.rs [hunk 2]
```diff
@@ -27,12 +27,19 @@ pub(crate) struct SpecCollector<'tcx, 'a> {
...
+
+/// Set of module (`LocalDefId`) that should be ignored by flux
+pub type Ignores = FxHashMap<IgnoreKey, ()>;
```
Comment (nilehmann) [inline flux-driver/src/collector.rs:34]: Make this a set

```suggestion
pub type Ignores = FxHashSet<IgnoreKey>;
```

File: flux-driver/src/callbacks.rs [hunk 3]
```diff
@@ -137,8 +162,13 @@ impl<'a, 'genv, 'tcx> CrateChecker<'a, 'genv, 'tcx> {
         let aliases = specs.aliases;
 
         // Function signatures
-        specs
+        let fn_specs: Vec<(LocalDefId, crate::collector::FnSpec)> = specs
             .fns
+            .into_iter()
+            .filter(|(def_id, _)| !is_ignored(&genv.tcx, &specs.ignores, def_id))
+            .collect();
```
Comment (nilehmann) [inline flux-driver/src/callbacks.rs:169]: I’d follow clippy suggestion here. You can either add this filter to the chain below or early return inside the `try_for_each_exhaust`

**Review by nilehmann — APPROVED** [review]
@ranjitjhala lgtm, just some stylistics comments

File: flux-driver/src/callbacks.rs (hunk 3 above)
Comment (ranjitjhala) [reply flux-driver/src/callbacks.rs:169]: Where do you see the clippy suggestion? (I had to do this split as the borrow checker was yelling about `tcx` being consumed by the filter...?

Comment (ranjitjhala) [reply flux-driver/src/callbacks.rs:169]: nvm pushed into the `try_for_each_exhaust`

**Pull request merged**